roles/java/tasks/main.yml

```diff
@@ -1,55 +1,32 @@
 ---
 # tasks file for roles/java
-- name: Check if dns entry is defined in /etc/resolv.conf
-  command:
-    cmd: 'grep -q 8.8.8.8 /etc/resolv.conf'
-  register: google_in_resolv
-  changed_when: false
-  failed_when: false
-
-- when: google_in_resolv.failed
-  name: fix dns issue with google dns
-  become: true
-  become_user: root
-  shell: 'echo "nameserver 8.8.8.8" > /etc/resolv.conf'
-
 - name: Download file with check (md5)
   get_url:
     url: "{{ dependencies_url.jdk }}"
     dest: "{{ download_location }}/{{ jdk_tar_file }}"
     checksum: "md5:{{ dependencies_url.jdk_md5_checksum }}"
     timeout: 570
 
-- name: Unarchive jdk tar.gz
-  unarchive:
-    src: "{{ download_location }}/{{ jdk_tar_file }}"
-    dest: "{{ download_location }}"
-    remote_src: true
-    mode: 'u=rwx,g=rx,o=rx'
-    creates: "{{ download_location }}/{{ jdk_folder }}/"
-
-- name: Copy Java
+- name: Install Java JRE {{ dependencies_version.jdk }}
   become: true
-  become_user: root
-  copy:
-    src: "{{ download_location }}/{{ jdk_folder }}/"
-    dest: "{{ java_home }}"
-    remote_src: true
-    owner: root
-    group: root
-    mode: 'u=rwx,g=rx,o=rx'
-
-- name: Check if setenv file exists
-  stat:
-    path: "{{ config_folder }}/setenv.sh"
-  register: setenv_exists
+  block:
+    - name: Unarchive jdk tar.gz
+      unarchive:
+        src: "{{ download_location }}/{{ jdk_tar_file }}"
+        dest: "{{ java_home | dirname }}"
+        remote_src: true
+        creates: "{{ java_home }}"
+        owner: root
+        group: root
+        extra_opts:
+          - --xform=s|[^/]*|{{ java_home | basename }}|
 
-- name: Add setenv bash file
-  become: true
-  template:
-    src: setenv.sh
-    dest: "{{ config_folder }}/setenv.sh"
-    owner: "{{ username }}"
-    group: "{{ group_name }}"
-    mode: 'u=rwx,g=rx,o=rx'
-  when: not setenv_exists.stat.exists
+    - name: Add setenv bash file
+      template:
+        src: setenv.sh
+        dest: "{{ config_folder }}/setenv.sh"
+        owner: "{{ username }}"
+        group: "{{ group_name }}"
+        mode: "0755"
+      tags:
+        - molecule-idempotence-notest
```

Comment:
if variables used in the template doesn't changes between run it should be idempotent 🤔

Reply:
see my previous comment. It's all about handling the same file in multiple roles